fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerUniqueIndexTest.java

```diff
@@ -366,4 +371,184 @@ public void resolveUniquenessViolations() {
+            }
+        } ;
+    }
+
+    private void disableAll(List<Index> indexes) {
+        try (FDBRecordContext context = openContext()) {
+            // disable all
+            for (Index index : indexes) {
+                recordStore.markIndexDisabled(index).join();
+            }
+            context.commit();
+        }
+    }
+
+    @Test
+    public void testUniquenessMultiTargetsForbidUniquePending() {
+        testUniquenessMultiTarget(false);
+    }
+
+    @Test
+    public void testUniquenessMultiTargetsAllowUniquePending() {
+        testUniquenessMultiTarget(true);
+    }
+
+    private void testUniquenessMultiTarget(boolean allow) {
+        List<TestRecords1Proto.MySimpleRecord> records = LongStream.range(0, 10).mapToObj( val ->
+                TestRecords1Proto.MySimpleRecord.newBuilder().setRecNo(val).setNumValue2(((int)val) % 5).build()
+        ).collect(Collectors.toList());
+
+        List<Index> indexes = new ArrayList<>();
+        indexes.add(new Index("indexA", field("num_value_2"), EmptyKeyExpression.EMPTY, IndexTypes.VALUE, IndexOptions.UNIQUE_OPTIONS));
+        indexes.add(new Index("indexB", field("num_value_3_indexed"), IndexTypes.VALUE));
+        indexes.add(new Index("indexC", field("num_value_unique"), EmptyKeyExpression.EMPTY, IndexTypes.VALUE, IndexOptions.UNIQUE_OPTIONS));
+        FDBRecordStoreTestBase.RecordMetaDataHook hook = allIndexesHook(indexes);
+
+        openSimpleMetaData();
+        try (FDBRecordContext context = openContext())  {
+            records.forEach(recordStore::saveRecord);
+            context.commit();
+        }
+        openSimpleMetaData(hook);
+        disableAll(indexes);
+        try (OnlineIndexer indexBuilder = OnlineIndexer.newBuilder()
+                .setDatabase(fdb).setMetaData(metaData).setSubspace(subspace)
+                .setTargetIndexes(indexes)
+                .setIndexingPolicy(OnlineIndexer.IndexingPolicy.newBuilder()
+                        .allowUniquePendingState(allow))
+                .build()) {
+            if (allow) {
+                indexBuilder.buildIndex();
+            } else {
+                RecordCoreException e = assertThrows(RecordCoreException.class, indexBuilder::buildIndex);
+                RuntimeException runE = FDBExceptions.wrapException(e);
+                assertNotNull(runE);
+                assertThat(runE, instanceOf(RecordIndexUniquenessViolation.class));
+            }
+        }
+        try (FDBRecordContext context = openContext()) {
+            // unique index with uniqueness violation:
+            if (allow) {
+                assertTrue(recordStore.isIndexReadableUniquePending(indexes.get(0)));
+            } else {
+                assertTrue(recordStore.isIndexWriteOnly(indexes.get(0)));
+            }
+            // non-unique index:
+            assertTrue(recordStore.isIndexReadable(indexes.get(1)));
+            // unique index with on unique number (no duplications):
```

Comment:
I'm guessing `on` should be something else here, but I'm not sure what

Reply:
Niether do I :) 
Changed to "unique index of unique numbers"